qualtran/bloqs/basic_gates/su2_rotation.py

```diff
@@ -148,7 +193,7 @@ def adjoint(self) -> 'SU2RotationGate':
         )
 
     def _t_complexity_(self) -> TComplexity:
-        return TComplexity(rotations=3)
+        return TComplexity(rotations=3, clifford=2)
```

Comment:
xref https://github.com/quantumlib/Qualtran/issues/1309#issuecomment-2299825657

do you want this to always be 3 rotations or do you want the angles to affect the complexity? Please implement build_call_graph, remove _t_complexity_ override, and add a test with `qlt_testing.assert_equivalent_bloq_counts`. If you want it always to be three rotations, return that in the call graph and use `generalize_rotation_angle` generalizer

Reply:
It was annotated with 3 rotations for simplicity, but I'll make it return the actual rotations so that classify_bloqs handles the special angles correctly.

Reply:
I removed the t-complexity override and simplified the decomposition to be Rz Rx Rz.

Should I also implement a call graph? because the decomposition here is always possible and small - 1-qubit 4-gates.

Reply:
it should be fine since the decomposition is small and supports symbolics